src/cmake-tools.ts

```diff
@@ -167,32 +166,25 @@ export class CMakeTools implements vscode.Disposable, api.CMakeToolsAPI {
    * Start up a new CMake driver and return it. This is so that the initialization
    * of the driver is atomic to those using it
    */
-  private async _startNewCMakeDriver(): Promise<CMakeDriver> {
+  private async _startNewCMakeDriver(cmake: CMakeExecutable): Promise<CMakeDriver> {
     const kit = this._kitManager.activeKit;
     log.debug('Starting CMake driver');
-    const cmake = await paths.cmakePath;
-    const version_ex = await proc.execute(cmake, ['--version']).result;
-    if (version_ex.retc !== 0 || !version_ex.stdout) {
-      throw new Error(`Bad CMake executable "${cmake}". Is it installed and a valid executable?`);
+    if (!cmake.isPresent) {
+      throw new Error(`Bad CMake executable "${cmake.path}".`);
     }
 
     let drv: CMakeDriver;
     if (config.useCMakeServer) {
-      console.assert(version_ex.stdout);
-      const version_re = /cmake version (.*?)\r?\n/;
-      const version = util.parseVersion(version_re.exec(version_ex.stdout)![1]);
-      // We purposefully exclude versions <3.7.1, which have some major CMake
-      // server bugs
-      if (util.versionGreater(version, '3.7.1')) {
-        drv = await CMakeServerClientDriver.create(this._stateManager, kit);
+      if (cmake.isServerModeSupported) {
+        drv = await CMakeServerClientDriver.create(cmake, this._stateManager, kit);
       } else {
-        log.info(
-            'CMake Server is not available with the current CMake executable. Please upgrade to CMake 3.7.2 or newer.');
-        drv = await LegacyCMakeDriver.create(this._stateManager, kit);
+        log.info(`CMake Server is not available with the current CMake executable. Please upgrade to CMake ${
+            cmake.minimalServerModeVersion} or newer.`);
```

Comment:
`cmake.minimalServerModeVersion` will not print correctly, since it's not a simple string. It will print as `[object Object]`.

Reply:
While here, can you change `log.info` to a warning? Just to make it more clear in the logs.